packages/cli/src/generators/index.js

```diff
@@ -0,0 +1,127 @@
+const path = require('path');
+
+const prettier = require('gulp-prettier');
+
+const {
+  NODE_VERSION,
+  PACKAGE_VERSION,
+  PLATFORM_PACKAGE
+} = require('../constants');
+
+const Generator = require('yeoman-generator');
+
+const AUTH_TYPE_CHOICES = [
+  {
+    name: 'Basic',
+    value: 'basic'
+  },
+  {
+    name: 'Custom (API Key)',
+    value: 'custom'
+  },
+  {
+    name: 'Digest',
+    value: 'digest'
+  },
+  {
+    name: 'OAuth1',
+    value: 'oauth1'
+  },
+  {
+    name: 'OAuth2',
+    value: 'oauth2'
+  },
+  {
+    name: 'Session',
+    value: 'session'
+  }
+];
+
+class ProjectGenerator extends Generator {
```

Comment:
Notice this class `ProjectGenerator` is in the `generators` directory. I use plural "generators" on purpose because I think we might want to add more sub-generators to power the `scaffold` command. For example, we could have a `TriggerGenerator` that powers `zapier scaffold trigger foo`.